### PR TITLE
#234: Make getScrollPosition @NonNullable

### LINKED ISSUE #234: Crash when drawable is null
We're getting reports about crashes in our app.

**See below stacktrace:**
> TouchImageView.java, line 527
> java.lang.NullPointerException: Attempt to read from field 'float android.graphics.PointF.x' on a null object reference
> com.ortiz.touchview.TouchImageView.setZoom TouchImageView.java:527
> com.ortiz.touchview.TouchImageView.setScaleType TouchImageView.java:230

Looking at the code, the crash makes sense. `getScrollPosition()` is actually @Nullable, so it's use inside `setZoom(img)` is not safe:

``` java
    public void setZoom(TouchImageView img) {
        PointF center = img.getScrollPosition();
       // `center` can AND will be null from time to time
        setZoom(img.getCurrentZoom(), center.x, center.y, img.getScaleType());
    }
```

We are loading images from the Gallery using Glide into the TouchImageView. Especially on low-end devices, this can result in a TouchImageView without a Drawable -> causing the crash. 

**To reproduce:**
Load a large image into the TouchImageView (+- 150mb).

### PR DESCRIPTION
**Pull request by Entreco**
This fixes #234 .

I'm not entirely sure what the best return value should be when `getDrawable()` returns null.

Based on the description "Return the point at the center of the zoomed image"., I decided to `return new PointF(.5F, .5F);` Not sure if that has undesired side effects.

### CODE REVIEW

File: touchview/src/main/java/com/ortiz/touchview/TouchImageView.java [hunk 1]
```diff
@@ -537,7 +537,7 @@ public void setZoom(TouchImageView img) {
     public PointF getScrollPosition() {
         Drawable drawable = getDrawable();
         if (drawable == null) {
-            return null;
+            return new PointF(.5F, .5F);
```
Comment (hannesa2) [inline touchview/src/main/java/com/ortiz/touchview/TouchImageView.java]: It's not wrong to center it, but is't it more restrict when we use `@NonNull` especailly when you read the comment ? 

```
/**
     * Set zoom parameters equal to another TouchImageView. Including scale, position,
     * and ScaleType.
     */
public void setZoom(@NonNull TouchImageView img) {
```
I've no strong opinion

Comment (Entreco) [reply touchview/src/main/java/com/ortiz/touchview/TouchImageView.java]: Not sure if I understand you correct, so let me rephrase.
The view is not the problem, it's the **drawable** of that view being null, causing the crash.

If you are suggesting to change `setZoom(TouchImageView img)` to `setZoom(@NonNull TouchImageView img)` -> that will **not** guarantee `getDrawable()` returning a NonNull value. So, that annotation will not prevent a crash.

I agree that enforcing a valid view seems the best option, but I don't have suggestions for enforcing a valid view. E.g. I have no ideas on how to enforce a view with a NonNull drawable.
